src/code/containers/challenge-container-selector.js

```diff
@@ -52,8 +56,10 @@ class ChallengeContainerSelector extends Component {
     if (routeParams.challengeId) {
       routeParams = AuthoringUtils.challengeIdToRouteParams(authoring, routeParams.challengeId);
     }
-    if (hasChangedRouteParams(this.props.currentRouteSpec, routeParams)) {
-      navigateToCurrentRoute({level: routeParams.level-1, mission: routeParams.mission-1, challenge: routeParams.challenge-1});
+    if (isValidRouteParams) {
```

Comment:
Alright, here are the changes we discussed @kswenson! Now, if route params are missing, we still navigate to a default level, and if we've already navigated to a level, we do nothing. 

FWIW, the `currentRouteSpec` when the challenge loads from a URL is set to 0.0.0